stable/external-dns/templates/deployment.yaml

```diff
@@ -0,0 +1,70 @@
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "external-dns.fullname" . }}
+  labels: {{ include "external-dns.labels" . | indent 4 }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
+spec:
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels: {{ include "external-dns.labels" . | indent 8 }}
+      {{- if .Values.podAnnotations }}
+      annotations:
+{{ toYaml .Values.podAnnotations | indent 8}}
+      {{- end }}
+    spec:
+      {{- if .Values.serviceAccountEnabled }}
+      serviceAccountName: {{ include "external-dns.sa-name" . }}
+      {{- end }}
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repository }}/{{.Values.image.name}}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        args:
+        - --provider={{ .Values.provider }}
+        - --policy={{ .Values.policy }}
+        - --registry={{ .Values.registry }}
+        - --txt-owner-id={{default .Release.Name .Values.txtOwnerId }}
+        {{- if .Values.compatibility }}
```

Comment:
Agreed, I prefer this approach. I've addressed all the other outstanding comments and will have a little tinker tonight to see if I can make this work and to pick a sane set of default options.